### ISSUE TITLE
CycleGAN: Using detach() on fake

### ISSUE THREAD

**Issue by morawi** [eriklindernoren/PyTorch-GAN#71]
Shouldn't `detach()` moved from to the fake we use as input to the buffer?
i.e., code should be changed from: 
```
fake_A_ = fake_A_buffer.push_and_pop(fake_A)
loss_fake = criterion_GAN(D_A(fake_A_.detach()), fake)

```
to 
```
fake_A_ = fake_A_buffer.push_and_pop(fake_A.detach())
loss_fake = criterion_GAN(D_A(fake_A_), fake)
```
This will help us save some memory; as it is just a buffer and the graph should not be retained in the buffer.
